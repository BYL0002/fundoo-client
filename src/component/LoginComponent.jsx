/**
 * @description Login Component
 * @author Yash
 * @since 26/11/2018
 * @version 1.1
 * @module component
 */
import React from 'react';
import { Redirect } from 'react-router-dom';
import { TextField, IconButton, InputAdornment, Button, Snackbar } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';
import { loginService } from '../service/UserService';
// import CustomizedSnackbars from './SnackBarTheme';

/**
 * @description LoginComponent class component for login
 */
class LoginComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            showpassword: false,
            redirectToReferrer: false,
            responseGot: false,
            snackOpen: false,
            snackMessage: "",
            snackBarVariant : ""
        }
        this.setValue = this.setValue.bind(this);
        this.handleShowPassword = this.handleShowPassword.bind(this);
        this.handleSnackClose = this.handleSnackClose.bind(this);
        this.loginUser = this.loginUser.bind(this);
    }

    /**
     * @description method to set value of state variables
     */
    setValue = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    }

    /**
     * @description method to set value of password to display or not
     */
    handleShowPassword() {
        this.setState({
            showpassword: !this.state.showpassword
        })
    }

    handleSnackClose = (event, reason) => {
        if (reason === 'clickaway') {
            return;
        }

        this.setState({ snackOpen: false });
    };

    loginUser() {

        if (this.state.email !== "") {
            if (this.state.password !== "") {

                let request = {
                    thread: "/login",
                    data: {
                        email: this.state.email,
                        password: this.state.password
                    }
                }
                loginService(request, (err, data) => {
                    if(err === false)
                    {
                        this.setState({
                            responseGot : data,
                            snackOpen : true,
                            snackMessage : 'Error Occured, Try Later'
                        })
                    }
                    else
                    {
                        this.setState({
                            responseGot : data
                        })
                    }
                })
            }
            else {
                this.setState({
                    snackOpen: true,
                    snackMessage: "Password Empty"
                });
            }
        }
        else {
            this.setState({
                snackOpen: true,
                snackMessage: "Email Empty"
            });
        }
    }

    render() {
        if (this.state.responseGot) return (<Redirect to="/dashboard" />)

        return (
            <div className="Form">
                <div className="formHeader">
                    Log in<span className="beforeDashboardTitle" >Fundoo Notes</span>
                </div>
                <div  >
                    <TextField  label={this.props.name} name="email" onChange={this.setValue} required className="emailTextFieldLoginPage"></TextField>
                </div>
                <div >
                    <TextField  className="passwordTextFieldLoginPage"  label="Password" type={this.state.showpassword ? 'text' : 'password'} required
                        value={this.state.password} onChange={this.setValue} name="password"
                        InputProps={{
                            endAdornment: (
                                <InputAdornment position="end">
                                    <IconButton aria-label="Toggle Password Visibility" onClick={this.handleShowPassword} >
                                        {this.state.showPassword ? <VisibilityOff /> : <Visibility />}
                                    </IconButton>
                                </InputAdornment>
                            )
                        }}
                    />
                </div>
                <div className = "forgotPasswordLinkLocationLoginPage" >
                    <a className="links" href="/forgotpassword">Forgot <b>Password</b></a>
                </div>

                <div >
                    <Button fullWidth onClick={this.loginUser} variant='extendedFab' color="primary"  className = "LoginSubmitButtons" >Login</Button></div>
                <div>
                    <span className="CenterTextStyle" >Don't have account? </span>
                    <a className="links" href="/register"> <b>Register</b> </a>
                </div>
                <Snackbar
                    anchorOrigin={{
                        vertical: 'bottom',
                        horizontal: 'left',
                    }}
                    open={this.state.snackOpen}
                    autoHideDuration={6000}
                    onClose={this.handleSnackClose}
                    ContentProps={{
                        'aria-describedby': 'message-id',
                    }}
                    color="primary"
                    variant = "success"
                    message={<span id="message-id">{this.state.snackMessage}</span>}
                    action={[
                        <IconButton key="close" aria-label="Close" color="inherit" onClick={this.handleSnackClose} >
                            <CloseIcon />
                        </IconButton>,
                    ]}
                />
            </div>
        )
    }
}

/**
 * @exports LoginComponent component so as screens can import it
 */
export default LoginComponent;