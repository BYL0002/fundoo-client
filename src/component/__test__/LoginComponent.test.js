import React from 'react';
import { shallow } from 'enzyme';
import LoginComponent from '../LoginComponent';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
describe('MyComponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<LoginComponent debug />);
  
    expect(component).toMatchSnapshot();
  });
});