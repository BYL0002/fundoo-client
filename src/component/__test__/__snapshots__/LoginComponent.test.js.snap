// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="Form"
>
  <div
    className="formHeader"
  >
    Log in
    <span
      className="beforeDashboardTitle"
    >
      Fundoo Notes
    </span>
  </div>
  <div>
    <TextField
      autoFocus={true}
      className="emailTextFieldLoginPage"
      helperText=""
      name="email"
      onChange={[Function]}
      required={true}
      select={false}
      variant="standard"
    />
  </div>
  <div>
    <TextField
      InputProps={
        Object {
          "endAdornment": <WithStyles(InputAdornment)
            position="end"
          >
            <WithStyles(IconButton)
              aria-label="Toggle Password Visibility"
              onClick={[Function]}
            >
              <pure(VisibilityIcon) />
            </WithStyles(IconButton)>
          </WithStyles(InputAdornment)>,
        }
      }
      className="passwordTextFieldLoginPage"
      helperText=""
      label="Password"
      name="password"
      onChange={[Function]}
      required={true}
      select={false}
      type="password"
      value=""
      variant="standard"
    />
  </div>
  <div
    className="forgotPasswordLinkLocationLoginPage"
  >
    <a
      className="links"
      href="/forgotpassword"
    >
      Forgot 
      <b>
        Password
      </b>
    </a>
  </div>
  <div>
    <WithStyles(Button)
      className="LoginSubmitButtons"
      color="primary"
      fullWidth={true}
      onClick={[Function]}
      variant="extendedFab"
    >
      Login
    </WithStyles(Button)>
  </div>
  <div>
    <span
      className="CenterTextStyle"
    >
      Don't have account? 
    </span>
    <a
      className="links"
      href="/register"
    >
       
      <b>
        Register
      </b>
       
    </a>
  </div>
  <e
    accessType="online"
    buttonText="Login"
    clientId="1094333084310-jjpvtc8dui543p8skrhhvotbe68rcenj.apps.googleusercontent.com"
    cookiePolicy="single_host_origin"
    disabledStyle={
      Object {
        "opacity": 0.6,
      }
    }
    fetchBasicProfile={true}
    icon={true}
    isSignedIn={false}
    jsSrc="https://apis.google.com/js/client:platform.js"
    onFailure={[Function]}
    onRequest={[Function]}
    onSuccess={[Function]}
    prompt=""
    scope="profile email"
    tag="button"
    theme="light"
    type="button"
    uxMode="popup"
  />
  <div
    className="g-signin2"
    data-onsuccess="onSignIn"
  />
  <WithStyles(Snackbar)
    ContentProps={
      Object {
        "aria-describedby": "message-id",
      }
    }
    action={
      Array [
        <WithStyles(IconButton)
          aria-label="Close"
          color="inherit"
          onClick={[Function]}
        >
          <pure(CloseIcon) />
        </WithStyles(IconButton)>,
      ]
    }
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    autoHideDuration={6000}
    color="primary"
    message={
      <span
        id="message-id"
      >
        
      </span>
    }
    onClose={[Function]}
    open={false}
    variant="success"
  />
</div>
`;
